utility.go

```diff
@@ -0,0 +1,86 @@
+package zgrab2
+
+import (
+	"errors"
+	"net"
+	"strings"
+
+	"github.com/ajholland/zflags"
+)
+
+var parser *flags.Parser
+
+func init() {
+	parser = flags.NewParser(&config, flags.Default)
+}
+
+func AddCommand(command string, shortDescription string, longDescription string, m Module) (*flags.Command, error) {
+	return parser.AddCommand(command, shortDescription, longDescription, m)
+}
+
+func ParseFlags() ([]string, error) {
+	r, err := parser.Parse()
+	if err == nil {
+		validateFrameworkConfiguration()
+	}
+	return r, err
+}
+
+// ParseInput takes input and parses it into either a list of IP addresses, domain name, or errors
+func ParseInput(s string) ([]net.IP, string, error) {
+	i := strings.IndexByte(s, ',')
+	j := strings.IndexByte(s, '/')
+
+	switch {
+	case i == -1 && j == -1:
+		//just ip or domain
+		if ip := net.ParseIP(s); ip != nil {
+			return []net.IP{ip}, "", nil
+		} else {
+			ips, err := net.LookupIP(s)
```

Comment:
@dadrian Currently I parse a cidr block and put it in a list. Bad idea. Trying to remove that and put it as a IPNet instead. I'm blocking on this line here, where I lookup the ip's associated with a domain. Should I just return 4 types, ([]net.IP, string, net.IPNet, error) or only perform a scan on the first IP associated with a domain? Presently zgrab2 will lookup a domain, find all IP's associated with that domain, and then scan all of them.

Reply:
Only scan the first one for now.